ta4j-core/src/main/java/org/ta4j/core/indicators/CriterionIndicator.java

```diff
@@ -0,0 +1,96 @@
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+package org.ta4j.core.indicators;
+
+import org.ta4j.core.AnalysisCriterion;
+import org.ta4j.core.BarSeries;
+import org.ta4j.core.BaseTradingRecord;
+import org.ta4j.core.TradingRecord;
+import org.ta4j.core.num.Num;
+
+/**
+ * Criterion indicator.
+ *
+ * <p>
+ * Transforrms an {@link AnalysisCriterion} to an indicator. Returns
+ * {@code true} if the calculated criterion value is better than or equal to a
+ * {@link #requiredCriterionValue}; otherwise, {@code false}.
+ */
+public final class CriterionIndicator extends CachedIndicator<Boolean> {
+
+    private final AnalysisCriterion criterion;
+    private final Num requiredCriterionValue;
+    private final TradingRecord tradingRecord;
+
+    /**
+     * Constructor.
+     *
+     * @param series                 the bar series
+     * @param tradingRecord          the trading record
+     * @param criterion              the criterion to get the calculated criterion
+     *                               value on a bar index
+     * @param requiredCriterionValue the required criterion value to test against
+     *                               the calculated value of {@code criterion}
+     */
+    public CriterionIndicator(BarSeries series, TradingRecord tradingRecord, AnalysisCriterion criterion,
+            Num requiredCriterionValue) {
+        super(series);
+        this.tradingRecord = tradingRecord;
+        this.criterion = criterion;
+        this.requiredCriterionValue = requiredCriterionValue;
+    }
+
+    @Override
+    protected Boolean calculate(int index) {
+        if (tradingRecord.getPositions().isEmpty()) {
+            return false;
+        }
+
+        var subTradingRecord = new BaseTradingRecord(tradingRecord.getStartingType(),
+                tradingRecord.getTransactionCostModel(), tradingRecord.getHoldingCostModel());
+
+        for (var position : tradingRecord.getPositions()) {
+            var entry = position.getEntry();
+            var exit = position.getExit();
+
+            // consider only those positions made till index
+            if (entry.getIndex() <= index) {
+                subTradingRecord.enter(entry.getIndex(), entry.getNetPrice(), entry.getAmount());
+            }
+            if (exit != null && exit.getIndex() <= index) {
+                subTradingRecord.exit(exit.getIndex(), exit.getNetPrice(), exit.getAmount());
+            }
+        }
+
+        var calculatedCriterionValue = criterion.calculate(getBarSeries(), subTradingRecord);
+        return criterion.betterThan(calculatedCriterionValue, requiredCriterionValue)
```

Comment:
why not returning the criterion value directly, instead of making this bit opinionated? i.e. let the caller decide what to do with the value